### PR TITLE
Feature/condo/sberdoma 1134/ticket assignment logic changes

### PR DESCRIPTION
**Pull request by AntonAL**
For `executor` field only those user are presented in options list, who has `canBeAssignedAsExecutor` ability.
For `assignee` field only those user are presented in options list, who has `canBeAssignedAsResponsible` ability.

### CODE REVIEW

**Review by dkoviazin — APPROVED** [review]

File: apps/condo/domains/ticket/utils/clientSchema/search.ts [hunk 1]
```diff
@@ -131,17 +137,24 @@ export async function searchTicketClassifier (client, value) {
...
+
+        const result = data.objs
+            .filter(withUser)
+            .filter(filter || Boolean)
+            .map(object => {
+                if (object.user) {
+                    return ({ text: object.name, value: object.user.id })
+                }
+            })
+        console.debug(result)
```
Comment (Dimitreee) [inline apps/condo/domains/ticket/utils/clientSchema/search.ts:156]: remove log

File: apps/condo/domains/ticket/utils/clientSchema/search.ts [hunk 2]
```diff
@@ -131,17 +137,24 @@ export async function searchTicketClassifier (client, value) {
...
     return async function (client, value) {
         const { data, error } = await _search(client, GET_ALL_ORGANIZATION_EMPLOYEE_QUERY, { value, organizationId })
         if (error) console.warn(error)
 
-        return data.objs.map(object => {
-            if (object.user) {
-                return ({ text: object.name, value: object.user.id })
-            }
-        }).filter(Boolean)
+        const withUser = ({ user }) => user
```
Comment (Dimitreee) [inline apps/condo/domains/ticket/utils/clientSchema/search.ts:146]: unnecessary local function

File: apps/condo/domains/ticket/components/BaseTicketForm/index.tsx [hunk 3]
```diff
@@ -221,6 +221,14 @@ const TicketPurpose = ({ validations, organizationId, disableUserInteraction })
         </UserNameField>
     )
 
+    const whoCanBeAssignedAsExecutor = ({ role }) => (
```
Comment (Dimitreee) [inline apps/condo/domains/ticket/components/BaseTicketForm/index.tsx:224]: unnecessary local function

File: apps/condo/domains/ticket/components/BaseTicketForm/index.tsx [hunk 4]
```diff
@@ -221,6 +221,14 @@ const TicketPurpose = ({ validations, organizationId, disableUserInteraction })
...
+        get(role, 'canBeAssignedAsExecutor', false)
+    )
+
+    const whoCanBeAssignedAsResponsible = ({ role }) => (
```
Comment (Dimitreee) [inline apps/condo/domains/ticket/components/BaseTicketForm/index.tsx:228]: unnecessary local function

File: apps/condo/domains/ticket/utils/clientSchema/search.ts (hunk 1 above)
Comment (AntonAL) [reply apps/condo/domains/ticket/utils/clientSchema/search.ts:156]: Removed in `3b3ff996`.

File: apps/condo/domains/ticket/utils/clientSchema/search.ts (hunk 2 above)
Comment (AntonAL) [reply apps/condo/domains/ticket/utils/clientSchema/search.ts]: Removed in `35fdb665`

File: apps/condo/domains/ticket/components/BaseTicketForm/index.tsx (hunk 3 above)
Comment (AntonAL) [reply apps/condo/domains/ticket/components/BaseTicketForm/index.tsx]: Removed in `35fdb665`

File: apps/condo/domains/ticket/components/BaseTicketForm/index.tsx (hunk 4 above)
Comment (AntonAL) [reply apps/condo/domains/ticket/components/BaseTicketForm/index.tsx]: Removed in `35fdb665`

File: apps/condo/domains/ticket/utils/clientSchema/search.ts (hunk 2 above)
Comment (AntonAL) [reply apps/condo/domains/ticket/utils/clientSchema/search.ts]: Removed in `2505cae1`

File: apps/condo/domains/ticket/components/BaseTicketForm/index.tsx [hunk 5]
```diff
@@ -254,7 +246,9 @@ const TicketAssignments = ({ validations, organizationId, propertyId, disableUse
                     >
                         <GraphQlSearchInput
                             formatLabel={formatUserFieldLabel}
-                            search={searchEmployeeUser(organizationId, whoCanBeAssignedAsExecutor)}
+                            search={searchEmployeeUser(organizationId, ({ role }) => (
+                                get(role, 'canBeAssignedAsExecutor', false)
```
Comment (pahaz) [inline apps/condo/domains/ticket/components/BaseTicketForm/index.tsx:250]: (not related to this task) could you please add the doc for `canBeAssignedAsExecutor` role field? It's not so clear to understand what's going on.

**Review by pahaz — APPROVED** [review]

File: apps/condo/domains/ticket/components/BaseTicketForm/index.tsx (hunk 5 above)
Comment (AntonAL) [reply apps/condo/domains/ticket/components/BaseTicketForm/index.tsx:250]: Added in `790eceb6`

**Pull request merged**